core/src/main/java/org/apache/iceberg/rest/RESTCatalog.java

```diff
@@ -63,13 +65,28 @@ public RESTCatalog() {
   }
 
   public RESTCatalog(Function<Map<String, String>, RESTClient> clientBuilder) {
-    this(SessionCatalog.SessionContext.createEmpty(), clientBuilder);
+    this(SessionCatalog.SessionContext.createEmpty(), clientBuilder, null, null);
   }
 
   public RESTCatalog(
       SessionCatalog.SessionContext context,
       Function<Map<String, String>, RESTClient> clientBuilder) {
-    this.sessionCatalog = new RESTSessionCatalog(clientBuilder, null);
+    this(context, clientBuilder, null, null);
+  }
+
+  public RESTCatalog(
+      Function<Map<String, String>, RESTClient> clientBuilder,
+      BiFunction<SessionCatalog.SessionContext, Map<String, String>, FileIO> ioBuilder,
+      RESTOperationsBuilder operationsBuilder) {
+    this(SessionCatalog.SessionContext.createEmpty(), clientBuilder, ioBuilder, operationsBuilder);
+  }
```

Comment:
Is this method necessary if we go with `this(SessionCatalog.SessionContext.createEmpty(), clientBuilder);` in line 68?

Reply:
Even though RESTSessionCatalog allows us to pass in the ioBuilder, but the RESTCatalog doesn't, so I add it in the constructor of RESTCatalog as well.